compiler/rustc_mir_build/src/build/expr/as_operand.rs

```diff
@@ -128,7 +128,8 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                 block.and(Operand::Constant(Box::new(constant)))
             }
             Category::Constant | Category::Place | Category::Rvalue(..) => {
-                let operand = unpack!(block = this.as_temp(block, scope, expr_id, Mutability::Mut));
+                let operand =
+                    this.as_temp(block, scope, expr_id, Mutability::Mut).unpack(&mut block);
```

Comment:
I have mixed feelings about this, the flow of `block` is easier to follow with the macro imo. Have we considered:
```suggestion
                    this.as_temp(&mut block, scope, expr_id, Mutability::Mut);
```

Reply:
I quite like using `&mut block` as an in+out parameter, and have used it in some of my previous coverage work.

The main reason I didn't try to use it here was to keep this PR as modest as possible. But it's certainly something I'd like to explore.